src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/subquery/SubQueryRewriter.java

```diff
@@ -0,0 +1,62 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.rewriter.subquery;
+
+import com.alibaba.druid.sql.ast.SQLExpr;
+import com.alibaba.druid.sql.ast.expr.SQLBinaryOpExpr;
+import com.alibaba.druid.sql.ast.expr.SQLExistsExpr;
+import com.alibaba.druid.sql.ast.expr.SQLInSubQueryExpr;
+import com.alibaba.druid.sql.ast.statement.SQLSelect;
+import com.alibaba.druid.sql.ast.statement.SQLSelectQuery;
+import com.alibaba.druid.sql.ast.statement.SQLTableSource;
+import com.alibaba.druid.sql.dialect.mysql.ast.statement.MySqlSelectQueryBlock;
+
+public class SubQueryRewriter {
```

Comment:
Not very clear about the role of this class, particularly **this Rewriter vs Rewriter interface vs RewriterRule**. I feel the functions here may be able to move to caller class.

Reply:
1) SubQueryRewriterRule is the entry point for SubQuery rewrite operation.Since our currently implementation has limitation, the match logic SubQueryRewriterRule could fast faill the SubQuery rewrite.
2) SubQueryRewriter traverse the AST and rewrite the Exists and In subquery.
3) Rewriter define the concerate Rewrite alg for each different case. e.g. NestedExistsRewriter will rewite the exists in nested query.